packages/twenty-server/src/modules/workflow/common/workspace-services/workflow-version.workspace-service.ts

```diff
@@ -0,0 +1,289 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+import { v4 } from 'uuid';
+
+import { TwentyORMManager } from 'src/engine/twenty-orm/twenty-orm.manager';
+import { WorkflowVersionWorkspaceEntity } from 'src/modules/workflow/common/standard-objects/workflow-version.workspace-entity';
+import {
+  WorkflowAction,
+  WorkflowActionType,
+} from 'src/modules/workflow/workflow-executor/workflow-actions/types/workflow-action.type';
+import { isDefined } from 'src/utils/is-defined';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { WorkflowBuilderWorkspaceService } from 'src/modules/workflow/workflow-builder/workflow-builder.workspace-service';
+import { ServerlessFunctionService } from 'src/engine/metadata-modules/serverless-function/serverless-function.service';
+import { WorkflowRecordCRUDType } from 'src/modules/workflow/workflow-executor/workflow-actions/record-crud/types/workflow-record-crud-action-input.type';
+import { WorkflowActionDTO } from 'src/engine/core-modules/workflow/dtos/workflow-step.dto';
+
+const TRIGGER_STEP_ID = 'trigger';
+
+@Injectable()
+export class WorkflowVersionWorkspaceService {
+  constructor(
+    private readonly twentyORMManager: TwentyORMManager,
+    private readonly workflowBuilderWorkspaceService: WorkflowBuilderWorkspaceService,
+    private readonly serverlessFunctionService: ServerlessFunctionService,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+  ) {}
+
+  private async getStepDefaultDefinition({
+    type,
+    workspaceId,
+  }: {
+    type: WorkflowActionType;
+    workspaceId: string;
+  }): Promise<WorkflowAction> {
+    const newStepId = v4();
+
+    switch (type) {
+      case WorkflowActionType.CODE: {
+        const newServerlessFunction =
+          await this.serverlessFunctionService.createOneServerlessFunction(
+            {
+              name: 'A Serverless Function Code Workflow Step',
+              description: '',
+            },
+            workspaceId,
+          );
+
+        if (!isDefined(newServerlessFunction)) {
+          throw new Error('Fail to create Code Step');
+        }
+
+        return {
+          id: newStepId,
+          name: 'Code - Serverless Function',
+          type,
+          valid: false,
+          settings: {
+            input: {
+              serverlessFunctionId: newServerlessFunction.id,
+              serverlessFunctionVersion: '',
+              serverlessFunctionInput: {},
+            },
+            outputSchema: {},
+            errorHandlingOptions: {
+              continueOnFailure: {
+                value: false,
+              },
+              retryOnFailure: {
+                value: false,
+              },
```

Comment:
I think we have default settings somewhere, to avoid rewriting:

`outputSchema: {},
            errorHandlingOptions: {
              continueOnFailure: {
                value: false,
              },
              retryOnFailure: {
                value: false,
              },`